src/python/pants/backend/codegen/protobuf/python/additional_fields.py

```diff
@@ -33,10 +33,15 @@ def rules():
     return [
         ProtobufSourceTarget.register_plugin_field(ProtobufPythonInterpreterConstraintsField),
         ProtobufSourcesGeneratorTarget.register_plugin_field(
-            ProtobufPythonInterpreterConstraintsField
+            ProtobufPythonInterpreterConstraintsField,
+            as_moved_field=True,
         ),
         ProtobufSourceTarget.register_plugin_field(ProtobufPythonResolveField),
-        ProtobufSourcesGeneratorTarget.register_plugin_field(ProtobufPythonResolveField),
+        ProtobufSourcesGeneratorTarget.register_plugin_field(
+            ProtobufPythonResolveField, as_moved_field=True
+        ),
```

Comment:
seeing this, I realize, why not just have a dedicated register method for the moved case, rather than overriding and introducing a flag?
```suggestion
        ProtobufSourcesGeneratorTarget.register_moved_plugin_field(
            ProtobufPythonResolveField,
        ),
```

Reply:
I like that ... but maybe a reservation is that it almost seems that `moved` _should_ be the default behavior of `register_plugin_field`. The migration path if we became confident would be:

- flag-based
  * make the default value = False
- new method based
  * make `register_plugin_field` behave like `register_moved_plugin_field`
  * deprecate `register_moved_plugin_field`
 
I am not sure either here or there. If we are not planning to change the default behavior of `register_plugin_field`, which perhaps is the best for strict backwards compatibility, then having the new method `register_moved_plugin_field` is way better than the flag.